modules/github_reverse_proxy/main.tf

```diff
@@ -0,0 +1,72 @@
+resource "null_resource" "zip_lambda_function" {
+  provisioner "local-exec" {
+    command = "zip -j lambda_function.zip lambda_function.py"
+  }
+
+  triggers = {
+    py_source = filemd5("lambda_function.py")
+  }
+}
+
+resource "aws_lambda_function" "github_webhook_handler" {
+  function_name = "${var.deployment_name}-github-webhook-handler"
+  role          = aws_iam_role.lambda_role.arn
+  handler       = "lambda_function.lambda_handler" # Python handler
+  runtime       = "python3.12"
+
+  filename         = "${path.module}/lambda_function.zip"
+  source_code_hash = filebase64sha256("${path.module}/lambda_function.zip")
+
+  environment {
+    variables = {
+      GITHUB_SECRET           = var.github_secret
+      PRIVATE_SYSTEM_ENDPOINT = var.private_system_endpoint
+    }
+  }
+
+  vpc_config {
+    subnet_ids         = var.vpc_private_subnets
+    security_group_ids = length(var.security_group_ids) > 0 ? var.security_group_ids : [aws_security_group.lambda_sg.id]
+  }
+
+  # Depend on the zip operation
+  depends_on = [null_resource.zip_lambda_function]
+}
+
+# API Gateway to act as a reverse proxy
+resource "aws_api_gateway_rest_api" "webhook_api" {
+  name = "GitHub Webhook Reverse Proxy"
```

Comment:
So interesting question. Is the API gateway capable of buffering the webhooks if the payload cannot be sent?

Reply:
I believe the API gateway will timeout after 29 seconds.

If we want something like that we should implement this fail-over ourselves. Which increases the complexity fast.